Facade types for N3.js and jsonld.js

betehess opened the pull request:
Also includes banana-rdf integration (`Future`-base `RDFReader`s).

betehess committed:
[N3.js] First cut at exposing N3.js API.

Can be "tested" with
  testOnly org.w3.banana.n3js.N3Test

betehess committed:
[N3.js] Async parsing w/ error handling (Future-s)

betehess committed:
[N3.js] Added support for prefixes.

betehess committed:
[N3.js] Use js.Dictionary instead of introspecting properties.

betehess committed:
[N3.js] Removed unwanted scalajs-dom stuff.

bblfish commented:
just had a quick look. To avoid having huge JS files in github, as we have with rdfstorew, it would be good to be able to import them as external jars just as one does with other external projects such as Jena or Sesame. I think the tool people like to use is [WebJars](http://www.webjars.org). Perhaps this is something @RubenVerborgh would do or know about or has already done?

RubenVerborgh commented:
While I do like N3.js integration, I would indeed prefer that they are imported as externals. This is the best guarantee of keeping them up to date.

I use npm as a package manager. Any chance this is compatible with your build process?

betehess commented:
@RubenVerborgh I can start looking at WebJars, see how it works, and come back to you.

betehess commented:
It's actually super easy: http://www.webjars.org/contributing

@RubenVerborgh, it's up to you if you want us to create/maintain the webjar repo for N3.js, or if you want to take care of it.

RubenVerborgh commented:
@betehess It would be great if you could do this, thanks. I get shivers from Maven.
Let me know if you need any help though.

betehess commented:
@RubenVerborgh it's almost ready. It would be nice if we could have the lib version as part of the URL. See https://github.com/betehess/org.webjars-N3.js/blob/master/pom.xml#L45 .

betehess commented:
Never mind, I made it work.

betehess commented:
Submitted for addition to WebJars https://github.com/webjars/webjars/issues/796

RubenVerborgh commented:
Very cool, thanks for this, @betehess!

betehess committed:
[N3.js] N3.js as an external dependency (WebJar)

betehess committed:
[N3.js] Moved facade types under a sub js package.

betehess committed:
[N3.js] Added a Future-friendly parse-by-chunk function.

betehess committed:
[N3.js] Implemented Banana-ready Turtle parser.

betehess committed:
[N3.js] Checks parser.addChunk is synchronous.

betehess committed:
[jsonld.js] Facade types for jsonld.js.

Important: the Javascript dependency is checked in because it had to be
modified to make sure that it worked in the Node.js environment. The
jsonld.js team is working on fixing that issue.
